src/tools/rbd_mirror/LeaderWatcher.h

```diff
@@ -0,0 +1,96 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_LEADER_WATCHER_H
+#define CEPH_RBD_MIRROR_LEADER_WATCHER_H
+
+#include "librbd/ManagedLock.h"
+#include "librbd/Watcher.h"
+#include "MirrorStatusWatcher.h"
+#include "tools/rbd_mirror/leader_watcher/Types.h"
+
+namespace librbd {
+  class ImageCtx;
+}
+
+namespace rbd {
+namespace mirror {
+
+class LeaderWatcher : public librbd::Watcher {
```

Comment:
Perhaps move ```librbd::Watcher``` to protected inheritence since ```init``` and ```shut_down``` should be invoked vs ```register_watch``` and ```unregister_watch```